pyfstat/make_sfts.py

```diff
@@ -441,50 +445,61 @@ def check_cached_data_okay_to_use(self, cl_mfd):
                     )
                 )
 
-        logging.info("Checking new commandline against existing SFT header...")
-        catalog = lalpulsar.SFTdataFind(self.sftfilepath, None)
-        cl_old = helper_functions.get_lalapps_commandline_from_SFTDescriptor(
-            catalog.data[0]
-        )
-        if len(cl_old) == 0:
-            logging.info(
-                "Could not obtain comparison commandline from old SFT header. "
-                + need_new
+        logging.info("...checking new commandline against existing SFT header(s)...")
+        # here we check one SFT header from each SFT file,
+        # assuming that any concatenated file has been sanely constructed with
+        # matching CLs
+        for sftfile in sftfiles:
+            catalog = lalpulsar.SFTdataFind(sftfile, None)
+            cl_old = helper_functions.get_lalapps_commandline_from_SFTDescriptor(
+                catalog.data[0]
             )
-            return False
-        if not helper_functions.match_commandlines(cl_old, cl_mfd):
-            logging.info("Commandlines unmatched. " + need_new)
-            return False
-        else:
-            logging.info("OK: Commandline matched with old SFT header.")
-        logging.info("Looks like cached data matches current options, will re-use it!")
+            if len(cl_old) == 0:
+                logging.info(
+                    "......ould not obtain comparison commandline from first SFT"
```

Comment:
I guess this is just a typo:
```suggestion
                    "......could not obtain comparison commandline from first SFT"
```